### PR TITLE
feat: Remove default values from URL

### LINKED ISSUE #656: remove default values from URL

### PR DESCRIPTION
**Pull request by murilx**
- Added the search middleware `stripSearchParams` to remove search params with default values from the URL
- Changed `from` options to use route instead of index e.g from `/tree/$treeId/` to `/tree/$treeId`
- Added some default clauses to zod types to make the default values of search params be returned when they are stripped from the URL. Without this clause the returned value would be undefined

Closes #656

## How to test
Navigate through the app, verifying that default values aren't being shown in the URL while keeping the same behaviour. Also note that when changing something, the search params goes back to the URL.

### CODE REVIEW

File: dashboard/src/types/tree/TreeDetails.tsx [hunk 1]
```diff
@@ -146,7 +163,8 @@ export const zTreeInformation = z
     commitName: z.string().optional().catch(''),
```
Comment (WilsonNet) [inline dashboard/src/types/tree/TreeDetails.tsx:163]: another ticket for you, create a ticket to remove parameters that does not need to be in the URL

the URL is to remember user choices

File: dashboard/src/routes/hardware/route.tsx [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
 
+const defaultValues = {
+  intervalInDays: DEFAULT_HARDWARE_INTERVAL_IN_DAYS,
+  hardwareSearch: '',
+};
+
 const zHardwareSchema = z.object({
   intervalInDays: makeZIntervalInDays(DEFAULT_HARDWARE_INTERVAL_IN_DAYS),
-  hardwareSearch: z.string().optional().catch(undefined),
+  hardwareSearch: z.string().optional().catch(''),
```
Comment (WilsonNet) [inline dashboard/src/routes/hardware/route.tsx:14]: from what I'm understanding with stripSerach poarameters you don't need `.optional` anymore

File: dashboard/src/routes/tree/route.tsx [hunk 3]
```diff
@@ -1,14 +1,21 @@
...
 import { DEFAULT_TIME_SEARCH } from '@/pages/treeConstants';
 
+const defaultValues = {
+  intervalInDays: DEFAULT_TIME_SEARCH,
+  treeSearch: '',
+};
+
 export const RootSearchSchema = z.object({
   intervalInDays: makeZIntervalInDays(DEFAULT_TIME_SEARCH),
+  treeSearch: z.string().optional().catch(''),
```
Comment (WilsonNet) [inline dashboard/src/routes/tree/route.tsx:15]: ditto about optional

File: dashboard/src/pages/TreeDetails/Tabs/Tests/TestsTab.tsx [hunk 4]
```diff
@@ -35,14 +35,14 @@ interface TestsTabProps {
...
-  const { treeId } = useParams({ from: '/tree/$treeId/' });
+  const { treeId } = useParams({ from: '/tree/$treeId' });
   const { isLoading, data, error } = useTreeDetails({
     treeId: treeId ?? '',
     filter: reqFilter,
   });
 
   const { tableFilter, diffFilter } = useSearch({
-    from: '/tree/$treeId/',
+    from: '/tree/$treeId',
```
Comment (WilsonNet) [inline dashboard/src/pages/TreeDetails/Tabs/Tests/TestsTab.tsx:45]: Why did you change this?
`/` is suppose to be the leaf, and that is the more exact place in this current route

File: dashboard/src/pages/TreeDetails/Tabs/Tests/TestsTab.tsx [hunk 5]
```diff
@@ -35,14 +35,14 @@ interface TestsTabProps {
 }
 
 const TestsTab = ({ reqFilter }: TestsTabProps): JSX.Element => {
-  const { treeId } = useParams({ from: '/tree/$treeId/' });
+  const { treeId } = useParams({ from: '/tree/$treeId' });
```
Comment (WilsonNet) [inline dashboard/src/pages/TreeDetails/Tabs/Tests/TestsTab.tsx:38]: ditto about leaf

File: dashboard/src/types/tree/TreeDetails.tsx (hunk 1 above)
Comment (murilx) [reply dashboard/src/types/tree/TreeDetails.tsx:163]: Done https://github.com/kernelci/dashboard/issues/671

**Review by MarceloRobert — COMMENTED** [review]
From my tests, it seems that the buildDetails page still receives an empty diffFilter, and the hardwareDetails page is behaving weirdly, I can't deselect trees and it seems laggy

File: dashboard/src/routes/hardware/route.tsx (hunk 2 above)
Comment (murilx) [reply dashboard/src/routes/hardware/route.tsx]: Done

**Review by WilsonNet — APPROVED** [review]
It is working much better than it was, maybe down the line we can make the urls more precise

**Review by MarceloRobert — COMMENTED** [review]
Going from a build details to a test details the default values are still showing - [example](http://localhost:5173/build/maestro%3A675ae0a0ca49c97d2997f2c4?intervalInDays=1&startTimestampInSeconds=1733943600&endTimestampInSeconds=1734030000)

**Review by MarceloRobert — APPROVED** [review]
LGTM, worked well on my tests
